server/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -56,6 +57,8 @@ public class ClusterService extends AbstractLifecycleComponent {
 
     private RerouteService rerouteService;
 
+    private AllocationService allocationService;
```

Comment:
I'm looking for suggestions if this can be made better. Right now the `allocationDeciders` are injected in the TransportActions that autoscaling uses, but I can't use injection since we made a design decision to use SPI for the reserved state handlers.

Essentially, I need a way to get to the allocationDeciders to be able to create/pass the validator in to the transport action transformation methods.